CMakeLists.txt

```diff
@@ -1,17 +1,54 @@
-cmake_minimum_required(VERSION 2.8)
+cmake_minimum_required(VERSION 2.8.11)
 
 project(Zopfli)
 
-option(BUILD_SHARED_LIBS "Build Zopfli with shared libraries" OFF)
+# Check if being built standalone
```

Comment:
Standalone build means to build in the source tree without a separate build directory, right?
I still haven't fully figure it out so please take my vote on making this check and it's use more obvious. Just my 2 cents.

Reply:
In CMake terminology, a build in the source tree is called an in-source build, and a build in a separate build directory an out-of-source build.

What I mean by standalone is that Zopfli is being built as the top-level (only) CMake project, as opposed to as a subproject of some other CMake based build.

An example of the latter would be [pngwolf-zopfli](https://github.com/jibsen/pngwolf-zopfli), which includes Zopfli as a git submodule. If Zopfli works as a CMake subproject, I would be able to do build and link it with
~~~.cmake
add_subdirectory(zopfli)
target_link_libraries(pngwolf PRIVATE libzopflipng)
~~~
rather than list all the source files inside the Zopfli directory.

Maybe the comment should be
~~~.cmake
# Check if being built standalone or as subproject
~~~

Reply:
That comment would help :+1: